File: ckgame-java/src/main/java/com/ckgame/politics/CouncilManager.java
package com.ckgame.politics;

import com.ckgame.world.World;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * 内阁注册表，按统治者管理内阁。
 * 对应 Python 的 CouncilRegistry。
 */
public final class CouncilManager {
    private final World world;
    private final Map<Integer, Council> byRuler = new HashMap<>();

    public CouncilManager(World world) {
        this.world = world;
    }

    public World world() {
        return world;
    }

    /** 获取或创建指定统治者的内阁。 */
    public Council councilFor(int ruler) {
        return byRuler.computeIfAbsent(ruler, k -> Council.empty(ruler));
    }

    /** 获取指定统治者的内阁；若不存在返回 null。 */
    public Council get(int ruler) {
        return byRuler.get(ruler);
    }

    /** 返回所有内阁的映射（供存档使用）。 */
    public Map<Integer, Council> allCouncils() {
        return byRuler;
    }

    /** 序列化内阁状态（供存档使用）。 */
    public Map<String, Object> saveState() {
        Map<String, Object> s = new HashMap<>();
        Map<String, Object> cm = new HashMap<>();
        for (var e : byRuler.entrySet()) {
            Council c = e.getValue();
            Map<String, Object> m = new HashMap<>();
            m.put("chancellor", c.chancellor);
            m.put("marshal", c.marshal);
            m.put("steward", c.steward);
            m.put("spymaster", c.spymaster);
            m.put("chaplain", c.chaplain);
            Map<String, String> tasks = new HashMap<>();
            for (var te : c.tasks.entrySet()) {
                tasks.put(te.getKey().name(), te.getValue().name());
            }
            m.put("tasks", tasks);
            cm.put(String.valueOf(e.getKey()), m);
        }
        s.put("councils", cm);
        return s;
    }

    /** 从存档恢复内阁状态。 */
    @SuppressWarnings("unchecked")
    public void loadState(Map<String, Object> s) {
        byRuler.clear();
        Map<String, Object> cm = (Map<String, Object>) s.get("councils");
        if (cm != null) {
            for (var e : cm.entrySet()) {
                int ruler = Integer.parseInt(e.getKey());
                Map<String, Object> m = (Map<String, Object>) e.getValue();
                Council c = Council.empty(ruler);
                c.chancellor = ((Number) m.get("chancellor")).intValue();
                c.marshal = ((Number) m.get("marshal")).intValue();
                c.steward = ((Number) m.get("steward")).intValue();
                c.spymaster = ((Number) m.get("spymaster")).intValue();
                c.chaplain = ((Number) m.get("chaplain")).intValue();
                Map<String, String> tasks = (Map<String, String>) m.get("tasks");
                if (tasks != null) {
                    for (var te : tasks.entrySet()) {
                        c.tasks.put(CouncilPosition.valueOf(te.getKey()),
                                CouncilTask.valueOf(te.getValue()));
                    }
                }
                byRuler.put(ruler, c);
            }
        }
    }
}

File: ckgame-ts/src/politics/CouncilManager.ts
import type { World } from '../world/World.js';
import { Council } from './Council.js';
import { CouncilPosition } from './CouncilPosition.js';
import { CouncilTask } from './CouncilTask.js';

/**
 * Council registry: manages councils per ruler.
 */
export class CouncilManager {
  private world: World | null;
  private byRuler: Map<number, Council> = new Map();

  constructor(world?: World | null) {
    this.world = world ?? null;
  }

  getWorld(): World | null {
    return this.world;
  }

  /** Get or create a council for the given ruler. */
  councilFor(ruler: number): Council {
    let c = this.byRuler.get(ruler);
    if (!c) {
      c = Council.empty(ruler);
      this.byRuler.set(ruler, c);
    }
    return c;
  }

  /** Get a ruler's council; returns undefined if none exists. */
  get(ruler: number): Council | undefined {
    return this.byRuler.get(ruler);
  }

  /** Return all councils. */
  allCouncils(): Map<number, Council> {
    return this.byRuler;
  }

  /** Serialize council state for save games. */
  saveState(): Record<string, unknown> {
    const s: Record<string, unknown> = {};
    const cm: Record<string, Record<string, unknown>> = {};
    for (const [ruler, c] of this.byRuler) {
      const m: Record<string, unknown> = {
        chancellor: c.chancellor,
        marshal: c.marshal,
        steward: c.steward,
        spymaster: c.spymaster,
        chaplain: c.chaplain,
      };
      const tasks: Record<string, string> = {};
      for (const [pos, task] of c.tasks) {
        tasks[pos] = task;
      }
      m['tasks'] = tasks;
      cm[String(ruler)] = m;
    }
    s['councils'] = cm;
    return s;
  }

  /** Restore council state from a save. */
  loadState(s: Record<string, unknown>): void {
    this.byRuler.clear();
    const cm = s['councils'] as Record<string, Record<string, unknown>> | undefined;
    if (!cm) return;
    for (const [key, m] of Object.entries(cm)) {
      const ruler = Number(key);
      const c = Council.empty(ruler);
      c.chancellor = Number(m['chancellor'] ?? -1);
      c.marshal = Number(m['marshal'] ?? -1);
      c.steward = Number(m['steward'] ?? -1);
      c.spymaster = Number(m['spymaster'] ?? -1);
      c.chaplain = Number(m['chaplain'] ?? -1);
      const tasks = m['tasks'] as Record<string, string> | undefined;
      if (tasks) {
        c.tasks.clear();
        for (const [posKey, taskKey] of Object.entries(tasks)) {
          c.tasks.set(posKey as CouncilPosition, taskKey as CouncilTask);
        }
      }
      this.byRuler.set(ruler, c);
    }
  }
}
